syntax = "proto3";

package ev_sales;

service EVSales {
  rpc GetSalesFiltered (SalesFilterRequest) returns (SalesReply);

  rpc UploadDataset (stream UploadRequest) returns (UploadStatus);
}

message UploadRequest {
  oneof payload {
    FileInfo info = 1;     
    bytes chunk_data = 2;   
  }
}

message FileInfo {
  string filename = 1;
}

message UploadStatus {
  bool success = 1;
  string message = 2;
}

message SalesFilterRequest {
  map<string, string> filters = 1;
}

message SalesReply {
  repeated string sales_xml = 1;
}